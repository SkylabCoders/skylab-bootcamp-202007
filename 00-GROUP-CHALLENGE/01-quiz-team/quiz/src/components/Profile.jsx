import React from "react";

function Profile(props){
    return (
        <p>USER PROFILE</p>
    );
}

export default Profile;