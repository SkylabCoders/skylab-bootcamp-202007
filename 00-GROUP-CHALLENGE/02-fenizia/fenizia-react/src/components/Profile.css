.profile-container {
  display: flex;
  flex-direction: column;
  margin: 2vh;
}
.profile-container__image {
  display: flex;
  flex-direction: column;
}
.profile-container__image__text {
  display: flex;
  flex-direction: column-reverse;
  padding: 8vh;
  font-weight: bold;
  font-size: larger;
}
.profile-container__image__text > p {
  text-align: center;
}
.profile-container__image__file {
  width: 25%;
  margin: auto;
}
.profile-container__image__file > img {
  width: 100%;
}
.profile-container__list__nav {
  display: flex;
  flex-direction: row;
  justify-content: center;
}
.profile-container__list__nav p {
  margin: 2vw;
}
.profile-container__list__nav a {
  border: 0.1rem solid transparent;
  background-color: #333;
  color: #fff;
  margin: 1vw;
  text-decoration: none;
  padding: 0.2rem 0.2rem 0.2rem 0.2rem;
  border-radius: 0.3rem;
  width: 20%;
  min-width: 190px;
  text-align: center;
}
.profile-container__list__nav a:hover {
  color: #fff;
  background-color: #ffcc00;
  border: 0.1rem solid #ffcc00;
  text-decoration: none;
  padding: 0.2rem 0.2rem 0.2rem 0.2rem;
  border-radius: 0.3rem;
}
/* QUERIES */

@media screen and (min-width: 1024px) {
  .profile-container__image__file {
    margin: 5vw;
  }
  .profile-container__list__nav {
    justify-content: start;
  }
  .profile-container__image {
    flex-direction: row;
  }
}
@media screen and (max-width: 599px) {
  .profile-container__list__nav a {
    min-width: 90px;
  }
}
