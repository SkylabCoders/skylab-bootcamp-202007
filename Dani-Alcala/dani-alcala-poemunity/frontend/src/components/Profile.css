.profile__main {
  height: 550px;
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  margin-top: 30px;
}

.profile__title {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  font-size: 24px;
  margin-bottom: 10px;
  padding-bottom: 10px;
  border-bottom: 2px solid #eee;
}

.profile__intro {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  height: 1600px;
}

.profile__image {
  display: inline-block;
  background-image: url(../icons/profile.svg);
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain;
  width: 35%;
  margin: 35px;
}

.profile__personal-data {
  width: 70%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
}

.profile__insert-poem {
  height: 70%;
}

.profile__insert-poem-title {
  font-size: 20px;
}

.profile__insert-poem-inputs {
  margin-bottom: 20px;
  margin-right: 10px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
  -webkit-box-pack: space-evenly;
      -ms-flex-pack: space-evenly;
          justify-content: space-evenly;
  width: 90%;
}

.profile__insert-poem-input {
  padding: .15rem;
  font-size: 1.2rem;
}

.profile__text-area {
  width: 70%;
  height: 230px;
  font-size: 24px;
  margin-left: 40px;
  z-index: 150;
}

.profile__send-poem {
  font-size: 22px;
  margin-top: 12px;
  margin-left: 40px;
  margin-bottom: 30px;
  padding: 6px;
  width: 15rem;
  background-color: red;
  color: white;
  cursor: pointer;
}

.profile__send-poem:hover {
  background-color: green;
  color: white;
}

.profile__myPoems {
  margin-top: -125px;
}

.profile__tabs {
  z-index: -100;
}

@media screen and (min-width: 700px) {
  .profile__title {
    font-size: 28px;
  }
  .profile__myPoems {
    margin-top: -72px;
  }
}

@media screen and (min-width: 900px) {
  .profile__title {
    font-size: 32px;
  }
  .profile__intro {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: horizontal;
    -webkit-box-direction: normal;
        -ms-flex-direction: row;
            flex-direction: row;
    height: 450px;
  }
  .profile__image {
    height: 65%;
    width: auto;
  }
  .profile__text-area {
    margin-left: 80px;
  }
  .profile__send-poem {
    margin-left: 80px;
  }
}
/*# sourceMappingURL=Profile.css.map */