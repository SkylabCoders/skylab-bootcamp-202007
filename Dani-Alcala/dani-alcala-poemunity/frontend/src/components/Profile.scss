.profile__main {
    height: 550px;
    width: 100%;
    display: flex;
    flex-direction: column;
    margin-top: 30px
}

.profile__title {
    display: flex;
    justify-content: center;
    font-size: 24px;
    margin-bottom: 10px;
    padding-bottom: 10px;
    border-bottom: 2px solid #eee;
    width: 80%;
    margin: auto
}

.profile__intro {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 1600px;
    margin-top: 10px;
    margin-left: 0px;
}

.profile__image {
    display: inline-block;
    background-image: url(../icons/profile.svg);
    background-position: center;
    background-repeat: no-repeat;
    background-size: contain;
    width: 35%;
    margin: 35px;
}

.profile__personal-data {
    width: 70%;
    display: flex;
    flex-direction: column;
}

.profile__insert-poem {
    height: 70%;
}

.profile__insert-poem-title {
    font-size: 20px;
}

.profile__insert-poem-inputs {
    margin-bottom: 20px;
    margin-right: 10px;
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    width: 90%;
}

.profile__insert-poem-input {
    padding: .15rem;
    font-size: 1.2rem;
}

.profile__text-area {
    width: 70%;
    height: 230px;
    font-size: 24px;
    margin-left: 40px;
    z-index: 150;
}

.profile__send-poem {
    font-size: 22px;
    margin-top: 12px;
    margin-left: 40px;
    margin-bottom: 30px;
    padding: 6px;
    width: 15rem;
    background-color: red;
    color: white;
    cursor: pointer;
}

.profile__send-poem:hover {
    background-color: green;
    color: white;
}

.profile__myPoems {
    margin-top: -125px
}

.profile__tabs {
    z-index: -100;
}

@media screen and (min-width: 700px) {
    .profile__title {
        font-size: 28px
    }

    .profile__myPoems {
        margin-top: -72px
    }
    
}

@media screen and (min-width: 900px) {
    .profile__title {
        font-size: 32px
    }

    .profile__intro {
        display: flex;
        flex-direction: row;
        height: 450px;
        margin-left: 10vw;
    }

    .profile__image {
        height: 65%;
        width: auto;
    }

    .profile__text-area {
        margin-left: 80px;
    }

    .profile__send-poem {
        
        margin-left: 80px;
        
    }
}