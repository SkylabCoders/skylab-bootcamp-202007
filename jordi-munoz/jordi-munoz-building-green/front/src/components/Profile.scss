.profile-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 80px;
  margin-bottom: 60px;
  text-align: center;
}
