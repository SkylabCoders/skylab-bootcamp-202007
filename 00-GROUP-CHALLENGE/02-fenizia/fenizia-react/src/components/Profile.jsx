import React, { useState } from "react";
import HistorialDePedidos from "./HistorialDePedidos";
import Favourite from "./Favourite";
import ProductosEnVenta from "./ProductosEnVenta";
import authStore from "../stores/authStore";
import userStore from '../stores/userStore'
import {} from "../user.mock";
import "./Profile.css";

function Profile() {
  const [showList, setShowList] = useState(<HistorialDePedidos />);
  const [user] = useState(checkUser(authStore.getUserProfile()));
 function checkUser(user){
  return!!userStore.getUserById(user.uid)? userStore.getUserById(user.uid): userStore.addUser(user);

 }

  return (
    <div className="profile-container">
      <div className="profile-container__image">
        <div className="profile-container__image__file">
          <img
            src={
              user.photo
            }
            alt={user.displayName}
          />
        </div>
        <div className="profile-container__image__text">
          <p> Welcome {user.name}!</p>
        </div>
      </div>
      <div className="profile-container__list">
        <div className="profile-container__list__nav">
          <a
            href="/profile"
            onClick={(event) => {
              event.preventDefault();
              setShowList(<HistorialDePedidos />);
            }}
          >
            <p>Historial de Pedidos</p>
          </a>
          <a
            href="/profile"
            onClick={(event) => {
              event.preventDefault();
              setShowList(<Favourite favouriteList={user.favourites}/>);
            }}
          >
            <p>Lista de favoritos</p>
          </a>
          <a
            href="/profile"
            onClick={(event) => {
              event.preventDefault();
              setShowList(<ProductosEnVenta />);
            }}
          >
            <p>Productos en venta</p>
          </a>
        </div>
        <div className="profile-container__list__render">{showList}</div>
      </div>
    </div>
  );
}

export default Profile;
