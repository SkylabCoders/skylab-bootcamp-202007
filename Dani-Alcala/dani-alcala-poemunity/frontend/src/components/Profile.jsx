import React, { useState } from "react";
import PropTypes from "prop-types";
import SwipeableViews from "react-swipeable-views";
import { makeStyles, useTheme } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Tabs from "@material-ui/core/Tabs";
import Tab from "@material-ui/core/Tab";
import Typography from "@material-ui/core/Typography";
import Box from "@material-ui/core/Box";
import "./Profile.scss";
import "../App.scss";
import { savePoem } from "../actions/poemActions";
import MyPoems from "./MyPoems";
import MyFavouritePoems from "./MyFavouritePoems";
import { useAuth0 } from "@auth0/auth0-react";
import CircularProgress from "./CircularIndeterminate";


function TabPanel(props) {
  const { children, value, index, ...other } = props;

  return (
    <div
      role="tabpanel"
      hidden={value !== index}
      id={`full-width-tabpanel-${index}`}
      aria-labelledby={`full-width-tab-${index}`}
      {...other}
    >
      {value === index && (
        <Box p={3}>
          <Typography>{children}</Typography>
        </Box>
      )}
    </div>
  );
}

TabPanel.propTypes = {
  children: PropTypes.node,
  index: PropTypes.any.isRequired,
  value: PropTypes.any.isRequired,
};

function a11yProps(index) {
  return {
    id: `full-width-tab-${index}`,
    "aria-controls": `full-width-tabpanel-${index}`,
  };
}

const useStyles = makeStyles((theme) => ({
  root: {
    backgroundColor: theme.palette.background.paper,
    width: 500,
  },
}));

export default function Profile(props) {
  const theme = useTheme();
  const [value, setValue] = React.useState(0);

  let [poemContent, setPoemContent] = useState("");
  let [poemTitle, setPoemTitle] = useState("");
  let [poemCategory, setPoemCategory] = useState("");

  const { user, isAuthenticated, isLoading } = useAuth0();

  if (isLoading) {
    return <CircularProgress />;
  }

  const handleChange = (event, newValue) => {
    setValue(newValue);
  };

  const handleChangeIndex = (index) => {
    setValue(index);
  };

  function onFieldChange(value, setValue) {
    setValue(value);
  }

  function handleSubmit(event) {
    event.preventDefault();
    event.target.reset();

    let current_datetime = new Date();
    let formattedDate =
      current_datetime.getFullYear() +
      "-" +
      (current_datetime.getMonth() + 1) +
      "-" +
      current_datetime.getDate() +
      " " +
      current_datetime.getHours() +
      ":" +
      current_datetime.getMinutes() +
      ":" +
      current_datetime.getSeconds();

    savePoem({
      userId: user.sub,
      author: user.name,
      picture: user.picture,
      poem: poemContent,
      title: poemTitle,
      genre: poemCategory,
      likes: [],
      date: formattedDate,
    });
    setPoemContent("");
    setPoemTitle("");
    setPoemCategory("");
  }

  return (
    isAuthenticated && (
      <main className="profile__main">
        <section className="profile__title">
          <div>Perfil de {user.name}</div>
        </section>
        <section className="profile__intro">
          <img
            className="profile__image"
            src={user.picture}
            alt={user.name}
          ></img>
          <div className="profile__personal-data">
            <div className="profile__insert-poem">
              <p className="profile__insert-poem-title">Insertar un poema:</p>
              <br></br>
              <form
                className="profile__insert-poem-form"
                onSubmit={handleSubmit}
              >
                <div className="profile__insert-poem-inputs">
                  <label className="profile__insert-poem-input">
                    {`Título: `}
                    <input
                      className="profile__insert-poem-input"
                      placeholder="Título"
                      name="title"
                      required
                      value={poemTitle}
                      onChange={(event) =>
                        onFieldChange(event.target.value, setPoemTitle)
                      }
                    />
                  </label>
                  <label className="profile__insert-poem-input">
                    {`Categoría: `}
                    <select
                      className="profile__insert-poem-input"
                      id="category"
                      name="category"
                      required
                      onChange={(event) => {
                        onFieldChange(event.target.value, setPoemCategory);
                      }
                      }
                    >
                      <option value="">Seleccionar</option>
                      <option value="alegria">Alegría</option>
                      <option value="amistad">Amistad</option>
                      <option value="amor">Amor</option>
                      <option value="aniversario">Aniversario</option>
                      <option value="desamor">Desamor</option>
                      <option value="haikus">Haikus</option>
                      <option value="infantiles">Infantiles</option>
                      <option value="tristes">Tristes</option>
                    </select>
                  </label>
                </div>
                <div>
                  <textarea
                    className="profile__text-area"
                    id="poem"
                    name="poem"
                    required
                    placeholder="Inserta aquí tu poema"
                    value={poemContent}
                    onChange={(event) =>
                      onFieldChange(event.target.value, setPoemContent)
                    }
                  ></textarea>
                </div>

                <button className="profile__send-poem" type="submit">
                  Enviar
                </button>
              </form>
            </div>
          </div>
        </section>
        <section className="profile__outro">
          <div className="profile__tabs">
            <AppBar position="static" color="default">
              <Tabs
                value={value}
                onChange={handleChange}
                indicatorColor="primary"
                textColor="primary"
                variant="fullWidth"
                aria-label="full width tabs example"
              >
                <Tab label="Mis poemas" {...a11yProps(0)} />
                <Tab label="Mis poemas favoritos" {...a11yProps(1)} />
              </Tabs>
            </AppBar>
            <SwipeableViews
              axis={theme.direction === "rtl" ? "x-reverse" : "x"}
              index={value}
              onChangeIndex={handleChangeIndex}
            >
              <TabPanel
                className="profile__myPoems"
                value={value}
                index={0}
                dir={theme.direction}
              >
                <MyPoems />
              </TabPanel>
              <TabPanel
                className="profile__myPoems"
                value={value}
                index={1}
                dir={theme.direction}
              >
                <MyFavouritePoems />
              </TabPanel>
            </SwipeableViews>
          </div>
        </section>
      </main>
    )
  );
}
