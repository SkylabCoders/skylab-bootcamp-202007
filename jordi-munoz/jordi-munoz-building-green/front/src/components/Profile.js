import React, { useEffect, useState } from "react";
import { useAuth0 } from "@auth0/auth0-react";
import { domain as DOMAIN } from '../auth0/auth0';
import './Profile.scss';
import LogoutButton from './logoutButton';
import { saveUser } from '../actions/authActions';
import authStore from '../stores/authStore';


const Profile = () => {
  const { user, isAuthenticated, getAccessTokenSilently } = useAuth0();
  const [userMetadata, setUserMetadata] = useState(null);

  useEffect(() => {

    const getUserMetadata = async () => {
      const domain = DOMAIN;

      try {
        const accessToken = await getAccessTokenSilently({
          audience: `https://${domain}/api/v2/`,
          scope: "read:current_user",
        });


        const userDetailsByIdUrl = `https://${domain}/api/v2/users/${user.sub}`;

        const metadataResponse = await fetch(userDetailsByIdUrl, {
          headers: {
            Authorization: `Bearer ${accessToken}`,
          },
        });

        const { user_metadata } = await metadataResponse.json();

        if (accessToken) {
          sessionStorage.setItem('USER_TOKEN', JSON.stringify(accessToken));
          saveUser(user);
          console.log(authStore.getUser());
          setUserMetadata(user_metadata);
        }
      } catch (e) {
        console.log(e.message);
      }
    };

    getUserMetadata();
  });



  return (
    isAuthenticated && (
      <div className="profile-container">
        <div >
          <img src={user.picture} alt={user.name} width="200px" />
          <h2>{user.name}</h2>
          <p>{user.email}</p>
        </div><br />
        <div style={{ width: "130px" }} >
          <LogoutButton />
        </div>
      </div>
    )
  );
};

export default Profile;