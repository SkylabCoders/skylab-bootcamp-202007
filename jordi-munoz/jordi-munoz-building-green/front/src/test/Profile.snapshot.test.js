import renderer from 'react-test-renderer';
import React from 'react';
import Profile from '../components/Profile';
import { BrowserRouter } from 'react-router-dom';

describe('profile snapshot', () => {
  const profileTree = renderer.create(
    <BrowserRouter>
      <Profile />
    </BrowserRouter>
  );
  it('should match', () => {
    expect(profileTree.toJSON()).toMatchSnapshot();
  });

});