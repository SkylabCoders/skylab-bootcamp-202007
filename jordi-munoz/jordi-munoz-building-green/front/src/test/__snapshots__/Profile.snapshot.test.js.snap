// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile snapshot should match 1`] = `null`;
